### ISSUE TITLE
Zoom in and out by scrolling is reversed with small images

### ISSUE THREAD

**Issue by Mark-64** [darktable-org/darktable#8014]
DT git master, both Linux and Windows
I noticed a strange behavior in darkroom when using "small" images, say below 2000x2000 (I don't know exactly the upper limit).

To reproduce:

- create and import small tiff image, for example 1000x1000
- go to darkroom
- try to zoom in and out with mouse scroll
- see the inverted behavior with respect to normal big images: scrolling up zooms out etc.

it is also irregular, sometimes it just stops scrolling

You can use the image attached (rename to .tif) or create one yourself

[DSC00185.tif.txt](https://github.com/darktable-org/darktable/files/5880390/DSC00185.tif.txt)

**Comment by Mark-64** [comment]
in darkroom.c, line 3653:
```
  if(up)
  {
    if((scale == 1.0f || scale == 2.0f) && !((state & GDK_CONTROL_MASK) == GDK_CONTROL_MASK)) return;
    if(scale >= 16.0f)
      return;
    else if(scale >= 8.0f)
      scale = 16.0;
    else if(scale >= 4.0f)
      scale = 8.0;
    else if(scale >= 2.0f)
      scale = 4.0;
    else if(scale < fitscale)
      scale += .05f * (1.0f - fitscale);
    else
      scale += .1f * (1.0f - fitscale);
  }
```
it is assumed that fitscale<1. This is not necessarily so, if the image is small and/or you have a HiDPI screen.
It fitscale>1, then the scale is decremented in zoom-in and the behavior is reversed. Same story for zoom out.

I will try to find a fix

**Comment by Mark-64** [comment]
Fixed with #8018
